### DISCUSSION TITLE
Effect of "Hide Loginname suffix"

### DISCUSSION THREAD

**Discussion by gabriel-hoff-vectorsolutions** [zitadel/zitadel#7328] [Q&A] [answered]
I have a use case where I know the user's organization beforehand, and I don't want users to have to enter their organization's domain as a suffix to their username.

Looking at [Zitadel's docs](https://zitadel.com/docs/apis/openidoauth/scopes), I found out you can configure it to automatically add the domain for you by passing `urn:zitadel:iam:org:domain:primary:domain` in the `scope` parameter. I tried it, and it works as expected:

![image](https://github.com/zitadel/zitadel/assets/62257776/63a298af-b234-4b4e-86d8-ea4a2f3fb1fc)

I then noticed a branding setting called "Hide Loginname suffix":

![image](https://github.com/zitadel/zitadel/assets/62257776/d766642a-5d31-4ef5-95ce-c6df575e3563)

I enabled this setting, expecting it to hide the domain suffix from the login page, but the domain continues to be presented.

My questions:
1. Am I correctly understanding the purpose of "Hide Loginname suffix", or could this be a bug?
2. Is there any way to hide that domain suffix when passing the `urn:zitadel:iam:org:domain:primary:domain` scope?

**Comment by gabriel-hoff-vectorsolutions** [comment]
FWIW, we don't expect our customers to care about their organization's domain, so hiding this domain would improve user experience by simplifying the information we are presenting them during login.

**Comment by hifabienne** [comment]
Hei @vs-gtadeu 
You are right the settings is exactly for what you want to do.
I quickly checked this myself and it worked perfectly fine for me.

You can set the settings once as default values on your instance.
And then you can overwrite the default setting on the specific organization.
Is the setting "Hide loginname suffix"  enabled on the default or on the organization?
And is it overwritten on the organization?

**Reply by gabriel-hoff-vectorsolutions** [reply]
> Is the setting "Hide loginname suffix" enabled on the default or on the organization?

The setting is only enabled for the organization. I'll try setting it at the instance level.

**Reply by gabriel-hoff-vectorsolutions** [reply, marked as answer]
I think I know what I did wrong here. I didn't notice the "Apply configuration" button, so I never pressed it:
![image](https://github.com/zitadel/zitadel/assets/62257776/96c5d82b-012d-403d-907f-16e9f2f2a9fe)

**Reply by gabriel-hoff-vectorsolutions** [reply]
@hifabienne, while we are here, would you mind answering another question?

When passing the `urn:zitadel:iam:org:domain:primary:domain` scope, I lose the ability to log in using an email address. Is that an intentional behavior?

![image](https://github.com/zitadel/zitadel/assets/62257776/32cfdb66-5094-4941-bef1-d8f40b320745)

**Reply by gabriel-hoff-vectorsolutions** [reply]
I decided to create https://github.com/zitadel/zitadel/discussions/7342 since it doesn't seem to be fully related to this question.
